layers/+lang/purescript/packages.el

```diff
@@ -24,22 +24,19 @@
   (add-hook 'purescript-mode-hook 'add-node-modules-path))
 
 (defun purescript/post-init-company ()
-  (when (configuration-layer/package-used-p 'psc-ide)
-    (spacemacs|add-company-backends
-      :backends company-psc-ide-backend
-      :modes purescript-mode)))
+  (add-hook 'purescript-mode-local-vars-hook #'spacemacs//purescript-setup-company))
 
 (defun purescript/post-init-flycheck ()
   (spacemacs/enable-flycheck 'purescript-mode))
 
 (defun purescript/init-purescript-mode ()
   (use-package purescript-mode
-    :defer t
     :init
     (progn
       (add-to-list 'spacemacs-indent-sensitive-modes 'purescript-mode)
       (add-hook 'purescript-mode-hook 'turn-on-purescript-indentation)
       (add-hook 'purescript-mode-hook 'purescript-decl-scan-mode)
+      (add-hook 'purescript-mode-local-vars-hook #'spacemacs//purescript-setup-backend)
```

Comment:
I would add this to `purescript-mode-hook` instead of the local vars hook, I am not sure that local vars hook is also executed when the buffer is restored from a previous editing session.